example/advection/advection_package.cpp

```diff
@@ -108,15 +109,16 @@ void PreFill(Container<Real> &rc) {
   int ie = pmb->ncells1 - 1;
   int je = pmb->ncells2 - 1;
   int ke = pmb->ncells3 - 1;
-  CellVariable<Real> &qin = rc.Get("advected");
-  CellVariable<Real> &qout = rc.Get("one_minus_advected");
-  for (int k = ks; k <= ke; k++) {
-    for (int j = js; j <= je; j++) {
-      for (int i = is; i <= ie; i++) {
-        qout(k, j, i) = 1.0 - qin(k, j, i);
-      }
-    }
-  }
+  PackIndexMap imap;
```

Comment:
Is a variable pack needed here if the variables aren't being iterated over? Does the original solution still work? Is this just for demonstration purposes?

Reply:
@jdolence wrote this bit. But my guess is it's not necessary and is just for demonstration purposes. You can of course use `ParArrayND.Get<3>()` and capture with the lambda.

Reply:
yes not necessary.  it's just meant to show that you can do things like, effectively, pass in a state vector and index into it as was originally done in athena++.